### PR TITLE
Unify map event gen1 gen2

### PR DESCRIPTION
**Pull request by KqesaR**
This PR is related to the proposal of @Rangi42 https://github.com/pret/pokered/issues/289

This PR is for the moment in WIP but you can review it to approve or not my contribution

### CODE REVIEW

**Comment by Rangi42** [comment]
Wow, thanks for taking the time to do this! I'm open to merging it or keeping the current macros, depending on what other pokered users think.

File: constants/hardware_constants.asm [hunk 1]
```diff
@@ -103,8 +103,8 @@ rLY         EQU $ff44 ; LCDC Y-Coordinate (R)
 rLYC        EQU $ff45 ; LY Compare (R/W)
 rDMA        EQU $ff46 ; DMA Transfer and Start Address (W)
 rBGP        EQU $ff47 ; BG Palette Data (R/W) - Non CGB Mode Only
-rOBP0       EQU $ff48 ; Object Palette 0 Data (R/W) - Non CGB Mode Only
```
Comment (Rangi42) [inline constants/hardware_constants.asm:106]: These "Object"s should stay as-is

File: data/events/hidden_objects.asm [hunk 2]
```diff
@@ -178,14 +178,14 @@ hidden_object: MACRO
 	db \2 ; y coord
 	db \1 ; x coord
 	db \3 ; item id
-	dba \4 ; object routine
+	dba \4 ; object_event routine
```
Comment (Rangi42) [inline data/events/hidden_objects.asm:181]: These "object"s should stay as-is

File: constants/hardware_constants.asm (hunk 1 above)
Comment (KqesaR) [reply constants/hardware_constants.asm:106]: Mistake :/ sorry i fix this

File: data/text/text_5.asm [hunk 3]
```diff
@@ -36,7 +36,7 @@ _FlashLightsAreaText::
 	prompt
 
 _WarpToLastPokemonCenterText::
-	text "Warp to the last"
+	text "warp_event to the last"
```
Comment (Rangi42) [inline data/text/text_5.asm:39]: This should definitely stay as-is; it's at least one reason why `make compare` is failing.

**Comment by KqesaR** [comment]
I have fix all mistake but i don't have finish the migration.

If it's ok for you @Rangi42 i can finish the migration (this PR is draft for the moment)

File: constants/hardware_constants.asm (hunk 1 above)
Comment (Rangi42) [reply constants/hardware_constants.asm]: Thanks!

**Comment by KqesaR** [comment]
I don't understand why the make compare failed because i have fixed all mistake in this PR

File: data/maps/objects/PalletTown.asm [hunk 4]
```diff
@@ -1,20 +1,20 @@
...
 	db $b ; border block
 
-	def_warps
-	warp  5,  5, 0, REDS_HOUSE_1F
-	warp 13,  5, 0, BLUES_HOUSE
-	warp 12, 11, 1, OAKS_LAB
+	def_warp_events
+	warp_event  5,  5, 0, REDS_HOUSE_1F, 0
+	warp_event 13,  5, 0, BLUES_HOUSE, 0
+	warp_event 12, 11, 1, OAKS_LAB, 1
```
Comment (dannye) [inline data/maps/objects/PalletTown.asm:7]: The warp events here have too many arguments.

File: data/maps/objects/PalletTown.asm [hunk 5]
```diff
@@ -1,20 +1,20 @@
...
+	bg_event 11,  5, 7 ; PalletTownText7
 
-	def_objects
-	object SPRITE_OAK, 8, 5, STAY, NONE, 1 ; person
-	object SPRITE_GIRL, 3, 8, WALK, ANY_DIR, 2 ; person
-	object SPRITE_FISHER, 11, 14, WALK, ANY_DIR, 3 ; person
+	def_object_events
+	object_event SPRITE_OAK, 8, 5, STAY, NONE, 1 ; person
+	object_event SPRITE_GIRL, 3, 8, WALK, ANY_DIR, 2 ; person
+	object_event SPRITE_FISHER, 11, 14, WALK, ANY_DIR, 3 ; person
```
Comment (dannye) [inline data/maps/objects/PalletTown.asm:18]: These object event arguments were not re-ordered.

File: data/maps/objects/Route16Gate1F.asm [hunk 6]
```diff
@@ -1,21 +1,21 @@
...
-	warp  0,  2, 4, LAST_MAP
-	warp  0,  3, 5, LAST_MAP
-	warp  7,  2, 6, LAST_MAP
-	warp  7,  3, 7, LAST_MAP
-	warp  6, 12, 0, ROUTE_16_GATE_2F
+	def_warp_events
+	warp_event  0,  8, LAST_MAP, 0
+	warp_event  0,  9, LAST_MAP, 1
+	warp_event  7,  8, LAST_MAP, 2
+	warp_event  7,  9, LAST_MAP, 3
```
Comment (dannye) [inline data/maps/objects/Route16Gate1F.asm:8]: Should be `warp_event  7,  9, LAST_MAP, 2`.

File: data/maps/objects/Route22.asm [hunk 7]
```diff
@@ -1,14 +1,14 @@
 Route22_Object:
 	db $2c ; border block
 
-	def_warps
-	warp  8,  5, 0, ROUTE_22_GATE
+	def_warp_events
+	warp_event  8,  5, ROUTE_22_GATE, 5
```
Comment (dannye) [inline data/maps/objects/Route22.asm:5]: Should be `warp_event  8,  5, ROUTE_22_GATE, 0`.

File: data/maps/objects/Route23.asm [hunk 8]
```diff
@@ -1,22 +1,22 @@
...
 	db $f ; border block
 
-	def_warps
-	warp  7, 139, 2, ROUTE_22_GATE
-	warp  8, 139, 3, ROUTE_22_GATE
-	warp  4, 31, 0, VICTORY_ROAD_1F
-	warp 14, 31, 1, VICTORY_ROAD_2F
+	def_warp_events
+	warp_event  7, 139, ROUTE_22_GATE, 2
+	warp_event  8, 139, ROUTE_22_GATE, 2
```
Comment (dannye) [inline data/maps/objects/Route23.asm:6]: Should be `warp_event  8, 139, ROUTE_22_GATE, 3`.

**Comment by KqesaR** [comment]
My PR can be review now, the ```make compare``` is ok @Rangi42 @dannye

**Comment by Rangi42** [comment]
Looks good to me! @dannye please squash-merge this if you agree it should go in.

**Pull request merged**